src/Abp/Extensions/DayOfWeekExtensions.cs

```
@@ -8,19 +9,80 @@ namespace Abp.Extensions
     public static class DayOfWeekExtensions
     {
         /// <summary>
-        /// Check if given <see cref="DayOfWeek"/> value is weekend.
+        /// Check if a given <see cref="DayOfWeek"/> value is weekend.
         /// </summary>
         public static bool IsWeekend(this DayOfWeek dayOfWeek)
         {
             return dayOfWeek.IsIn(DayOfWeek.Saturday, DayOfWeek.Sunday);
         }
 
         /// <summary>
-        /// Check if given <see cref="DayOfWeek"/> value is weekday.
+        /// Check if a given <see cref="DayOfWeek"/> value is weekday.
         /// </summary>
         public static bool IsWeekday(this DayOfWeek dayOfWeek)
         {
             return dayOfWeek.IsIn(DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday);
         }
+
+        /// <summary>
+        /// Finds the NTH week day of a month.
+        /// </summary>
+        /// <param name="dayOfWeek">The day of week.</param>
+        /// <param name="year">The year.</param>
+        /// <param name="month">The month.</param>
+        /// <param name="n">The nth instance.</param>
+        /// <remarks>Compensates for 4th and 5th DayOfWeek of Month</remarks>
+        public static DateTime FindNthWeekDayOfMonth(this DayOfWeek dayOfWeek, int year, int month, int n)
+        {
+            var y = 0;
+
+            if (n > 5)
+            {
+                n = 5;
+            }
+
+            if (n < 1)
+            {
+                n = 1;
```

It is better to throw an exception here I think.